pyvista/plotting/_property.py

```diff
@@ -0,0 +1,318 @@
+        return self.GetOpacity()
+
+    @opacity.setter
+    def opacity(self, value):
+        if value is None:
+            return
+        self.SetOpacity(value)
+
+    @property
+    def show_edges(self):
+        """Return or set show edges."""
+        return self.GetEdgeVisibility()
+
+    @show_edges.setter
+    def show_edges(self, value):
+        if value is None:
+            value = self._theme.show_edges
+        self.SetEdgeVisibility(value)
+
+    @property
+    def lighting(self) -> bool:
+        """Return or set lighting."""
+        return self.SetLighting()
+
+    @lighting.setter
+    def lighting(self, value):
+        if value is None:
+            value = self._theme.lighting
+        self.SetLighting(value)
+
+    @property
+    def ambient(self):
+        """Return or set ambient."""
+        return self.GetAmbient()
+
+    @ambient.setter
+    def ambient(self, new_ambient):
+        self.SetAmbient(new_ambient)
+
+    @property
+    def diffuse(self):
+        """Return or set diffuse."""
+        return self.GetDiffuse()
+
+    @diffuse.setter
+    def diffuse(self, new_diffuse):
+        self.SetDiffuse(new_diffuse)
+
+    @property
+    def specular(self):
+        """Return or set specular."""
+        return self.GetSpecular()
+
+    @specular.setter
+    def specular(self, new_specular):
+        self.SetSpecular(new_specular)
+
+    @property
+    def specular_power(self):
+        """Return or set specular power."""
+        return self.GetSpecularPower()
+
+    @specular_power.setter
+    def specular_power(self, new_specular_power):
+        self.SetSpecularPower(new_specular_power)
+
+    @property
+    def metallic(self):
+        """Return or set metallic."""
+        _check_supports_pbr()
+        return self.GetMetallic()
+
+    @metallic.setter
+    def metallic(self, new_metallic):
+        _check_supports_pbr()
+        self.SetMetallic(new_metallic)
+
+    @property
+    def roughness(self):
+        """Return or set roughness."""
+        _check_supports_pbr()
+        return self.GetRoughness()
+
+    @roughness.setter
+    def roughness(self, new_roughness):
+        _check_supports_pbr()
+        self.SetRoughness(new_roughness)
+
+    @property
+    def interpolation(self) -> str:
+        """Return or set show edges of this property."""
+        return self.GetInterpolationAsString()
+
+    @interpolation.setter
+    def interpolation(self, new_interpolation):
+        if new_interpolation == 'Physically based rendering':
+            _check_supports_pbr()
+
+            self.SetInterpolationToPBR()
+        elif new_interpolation == 'Phong':
+            self.SetInterpolationToPhong()
+        elif new_interpolation == 'Gouraud':
+            self.SetInterpolationToGouraud()
+        elif new_interpolation == 'Flat' or new_interpolation is None:
+            self.SetInterpolationToFlat()
+        else:
+            raise ValueError(f'Invalid interpolation "{new_interpolation}"')
```

Comment:
Aside from the `'pbr'` vs `'Physically based rendering'` duality, one might also want to use `'phong'` instead of `'Phong'` etc., or try `'Physically-based rendering'`. This can be a moderate pain in the butt for users. I'd prefer to potentially normalize to lowercase (and converting dashes to spaces). This would actually be the perfect use case for an `Enum`, but I also don't want to overengineer this already 3.7k LOC PR.

Reply:
> 3.7k LOC PR.

Typical Alex PR.

Reply:
Done in [498d644](https://github.com/pyvista/pyvista/pull/2941/commits/498d6443c07c7ba843bef026c960c7847028a31a).